apps/mobile/src/features/accounts/balances/hooks/use-stx-balances.ts

```diff
@@ -0,0 +1,36 @@
+import { QueryObserverResult } from '@tanstack/react-query';
+
+import { Money, StxCryptoAssetBalance } from '@leather.io/models';
+import { useCryptoCurrencyMarketDataMeanAverage } from '@leather.io/query';
+import { baseCurrencyAmountInQuote, createMoney } from '@leather.io/utils';
+
+import { useStxBalancesQueries } from './use-stx-balances.query';
+
+interface StxBalance {
+  stxBalance: Money;
+  stxBalanceUsd: Money;
+}
+
+interface StxBalances {
+  totalStxBalance: Money;
+  accounts: QueryObserverResult<StxCryptoAssetBalance, unknown>[];
+}
+
+function useGetStxBalanceByAddresses(addresses: string[]): StxBalances {
+  const { totalData, queries } = useStxBalancesQueries(addresses);
+
+  if (!addresses || addresses.length === 0)
+    return { totalStxBalance: createMoney(0, 'STX'), accounts: [] };
+
+  return {
+    totalStxBalance: totalData,
+    accounts: queries, // TODO as yet unused but perhaps useful
```

Comment:
This `accounts` contains the pre-aggregated individual account balance data. 

I'm not sure yet if this is useful here. Perhaps it is to show account specific data on account cards.

We can also use the same query to call `useGetStxBalanceByAddresses` passing a single address if we want instead